### PR TITLE
Add Apache 2 license file

### PR DESCRIPTION
**Pull request by cjh1**

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+      License. However, in accepting such obligations, You may act only
+      on Your own behalf and on Your sole responsibility, not on behalf
+      of any other Contributor, and only if You agree to indemnify,
+      defend, and hold each Contributor harmless for any liability
+      incurred by, or claims asserted against, such Contributor by reason
+      of your accepting any such warranty or additional liability.
+
+   END OF TERMS AND CONDITIONS
+
+   APPENDIX: How to apply the Apache License to your work.
```
Comment (aashish24) [inline LICENSE]: @cjh1 you don't need the appendix part.

Comment (cjh1) [inline LICENSE]: True, but it doesn't hurt.

Comment (aashish24) [inline LICENSE]: Upto you :smile:  but this part should go into README.md or something (once you replace the company name, year etc). Just my 2 cents.

Comment (cjh1) [inline LICENSE]: This part should go at the top of every file ...

Comment (aashish24) [inline LICENSE]: thats a good point. i don't think we are following this, we should though. I will bring it up for some tools as well.

**Pull request merged**